plasmapy/analytic/parameters.py

```diff
@@ -0,0 +1,852 @@
+gyration, while Larmor frequency indicates that this frequency was
+discovered by Larmor.
+
+The docstrings for plasma parameter methods should define these
+quantities in ways that are understandable to students who are
+taking their first course in plasma physics and yet are useful to
+experienced plasma physicists.
+
+Unit conversions involving angles must be treated with care.  Angles
+are dimensionless but do have units.  Angular velocity is often
+given in units of radians per second, though dimensionally this is
+equivalent to inverse seconds.  Astropy will treat radians
+dimensionlessly when using the dimensionless_angles() equivalency,
+but dimensionless_angles() does not account for multiplicative
+factor of 2*pi that is used when converting between frequency (1 /
+s) and angular velocity (rad / s).  A clear way to do this
+conversion is to set up an equivalency between cycles/s and Hz:
+
+  >>> f_ce = omega_ce.to(u.Hz, equivalencies=[(u.cy/u.s, u.Hz)])
+
+However, dimensionless_angles() does work when dividing a velocity
+by an angular frequency to get a length scale:
+
+  >>> d_i = (c/omega_pi).to(u.m, equivalencies=dimensionless_angles())
+
+"""
+
+
+def Alfven_speed(B, density, ion="p"):
+    """Returns the Alfven speed.
+
+    Parameters
+    ----------
+    B : Quantity
+        The magnetic field in units such as Tesla or Gauss
+    density: Quantity
+        Either the ion number density in units such as 1 / m**3, or the
+        mass density in units such as kg / m**3
+    ion : string, optional
+        Representation of the ion species.  If not given, then the ions
+        are assumed to be protons.
+
+    Returns
+    -------
+    V_A : Quantity with units of velocity
+        The Alfven velocity of the plasma in units of meters per second.
+
+    Raises
+    ------
+    UnitConversionError:
+        If the magnetic field or density is not in correct units
+
+    UserWarning:
+        If the Alfven velocity exceeds the speed of light
+
+    Notes
+    -----
+    The Alfven velocity :math:`V_A` is the typical propagation speed
+    of magnetic disturbances in a plasma, and is given by:
+
+    .. math::
+    V_A = \frac{B}{\sqrt{\mu_0\rho}}
+
+    where the mass density is :math:`\rho \approx n_i m_i`.
+
+    This expression does not account for relativisitic effects.
+
+    Examples
+    --------
+
+    >>> from astropy import units as u
+    >>> import plasmapy
+    >>> B = 140 * u.G
+    >>> n = 5e19 * u.m**-3
+    >>> rho = n * plasmapy.constants.m_i
+    >>> element = 'H'
+    >>> V_A1 = Alfven_speed(B, n, element)
+    >>> print(V_A1)
+    43185.62686969444 m / s
+    >>> V_A2 = Alfven_speed(B, rho, element)
+    >>> print(V_A2)
+    43185.62686969444 m / s
+    >>> print(V_A2.cgs)
+    4318562.686969444 cm / s
+    >>> print(V_A2.to(u.km/u.s))
+    43.18562686969444 km / s
+
+    """
+
+    try:
+        if B.si.unit in ['1 / m3', 'kg / m3'] and density.si.unit in ['T']:
+            B, density = density, B
```

Comment:
This is interesting - the intention is to have interchangeable parameter ordering, right? I suppose it's a possibility now that we have `Quantities`, but this might mess up with static type checkers or something. Still, I'd keep it in - gotta experiment with them new possibilities!!